### PR TITLE
Implement a few C ext functions and macros to compile msgpack

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: lib/mri/mkmf.rb [hunk 1]
```diff
@@ -552,7 +552,7 @@ def try_link0(src, opt="", *opts, &b) # :nodoc:
       end
     else
       try_do(src, cmd, *opts, &b)
-    end and File.executable?(CONFTEST+$EXEEXT)
+    end and File.exist?(CONFTEST+$EXEEXT) # TruffleRuby: TRY_LINK generates a non-executable file containing bitcode
```
Comment (eregon) [inline lib/mri/mkmf.rb]: @bjfish This seems a proper fix for try_link.
Alternatively we could try to not pass `-emit-llvm` for TRY_LINK and compile a real executable as this is mostly used to check if library macros/methods are defined.

**Comment by chrisseaton** [comment]
Did this not get any specs passing?

**Comment by eregon** [comment]
I didn't try to untag any spec, just be able to `gem install msgpack`.
Should I try to untag some specs?

**Comment by chrisseaton** [comment]
If no extra specs passed that's fine - there may not be any. But if you're adding methods I'd definitely try an untag.

**Comment by eregon** [comment]
None of these new macros/functions have specs, but I fixed specs to make
```bash
jt untag -fs --sulong :capi 
```
work without problems.
I fixed rb_thread_call_without_gvl on the way, and it seems some rb_yield + break spec also passes now.

**Comment by pitr-ch** [comment]
Didn't know anybody else is working on any cext at the moment. Seems like some of the fixes I already had in zlib branch sorry for letting it lay there.

**Comment by eregon** [comment]
I was just curious trying it out on the latest GraalVM and figure out what it'd take to have msgpack compiling.

**Pull request merged**